x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/get_active_instances_overview.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import {
+  FAAS_BILLED_DURATION,
+  FAAS_DURATION,
+  FAAS_NAME,
+  METRIC_SYSTEM_FREE_MEMORY,
+  METRIC_SYSTEM_TOTAL_MEMORY,
+  SERVICE_NAME,
+  SERVICE_NODE_NAME,
+} from '../../../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../../../common/utils/environment_query';
+import { Coordinate } from '../../../../../typings/timeseries';
+import { getBucketSize } from '../../../../lib/helpers/get_bucket_size';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { calcMemoryUsed } from './helper';
+
+interface ActiveInstanceTimeseries {
+  serverlessDuration: Coordinate[];
+  billedDuration: Coordinate[];
+  avgMemoryUsed: Coordinate[];
+  memorySize: Coordinate[];
+}
+
+export interface ActiveInstanceOverview {
+  activeInstanceName: string;
+  serverlessFunctionName: string;
+  timeseries: ActiveInstanceTimeseries;
+  serverlessDurationAvg: number | null;
+  billedDurationAvg: number | null;
+  avgMemoryUsed: number | null;
+  memorySize: number | null;
+}
+
+export async function getServerlessActiveInstancesOverview({
```

Comment:
We could use two requests for this (main statistics and details statistics) to improve loading times. The second one can accept a list of instance IDs that are visible on the table and return the timeseries as we do in other pages (e.g. instances table on the service overview page).

Reply:
@gbamparop I don't think we'll have a great perf improvement in this case. For now, we won't have comparison on this page, Which was the main reason why we decided to have the `details statistics api`. WDYT?

Reply:
It depends on how many instances we expect to have in the clusters and the response can also be quite large with many instances. We could have some test data to see the loading times with both but can also be done at a later point